spec/controllers/queue_items_controller_spec.rb

```diff
@@ -0,0 +1,104 @@
+require 'rails_helper'
+
+describe QueueItemsController do
+
+  describe "GET index" do
+    it "redirects to the sign in page for unauthenticated users" do
+      get :index
+      expect(response).to redirect_to sign_in_path
+    end
+    it "renders the index page" do
+      session[:user_id] = Fabricate(:user).id
+      get :index
+      expect(response).to render_template(:index)
+    end
+    it "sets the @queue_items variable" do
+      session[:user_id] = Fabricate(:user).id
+      video = Fabricate(:video)
+      queue_items = Fabricate(:queue_item, video_id: video.id, user_id: session[:user_id])
+      get :index
+      expect(assigns(:queue_items)).to eq([queue_items])
+    end
+  end
+
+  describe "POST create" do
+    context "signed in" do 
+      before do
+        session[:user_id] = Fabricate(:user).id
+      end
+      it "creates a queue item" do
+        video = Fabricate(:video)
+        post :create, video_id: video.id
+        expect(QueueItem.count).to eq(1)
+      end
```

Comment:
There should be definitely an empty space between specs. Helps readability a lot.